horovod/common/controller.cc

```diff
@@ -628,53 +619,45 @@ void Controller::CoordinateCacheAndState(CacheCoordinator& cache_coordinator) {
                                (Request::RequestType)response.response_type());
     }
 
-    // End negotation phase for synced cache hit set entries.
+    // End negotiation phase for synced cache hit set entries.
     for (auto bit : cache_coordinator.cache_hits()) {
       auto& response = response_cache_.peek_response(bit);
       timeline_.NegotiateEnd(response.tensor_names()[0]);
     }
   }
 }
 
-ResponseList Controller::FuseResponses(std::deque<Response>& responses,
-                                       bool joined) {
+ResponseList Controller::FuseResponses(std::deque<Response>& responses) {
   ResponseList response_list;
   while (!responses.empty()) {
 
     auto response = responses.front();
     assert(response.tensor_names().size() == 1);
     responses.pop_front();
     int64_t tensor_size = 0;
-    DataType dtype;
     if (response.response_type() == Response::ResponseType::ALLREDUCE ||
         response.response_type() == Response::ResponseType::ADASUM) {
       // Attempt to add more responses to this fused response.
 
+      tensor_size = response.tensor_sizes()[0];
       std::deque<Response> skipped_responses;
       int64_t skipped_size = 0;
       while (!responses.empty()) {
         auto new_response = responses.front();
         assert(new_response.tensor_names().size() == 1);
 
-        std::vector<TensorTableEntry> entries_for_join;
-        if (joined) {
-          tensor_queue_.GetTensorEntriesFromResponse(response, entries_for_join,
-                                                     joined);
-        }
-
-        const auto& new_entry =
-            joined
-                ? entries_for_join[0]
-                : tensor_queue_.GetTensorEntry(new_response.tensor_names()[0]);
-        int64_t new_tensor_size = new_entry.tensor->size();
-
+        int64_t new_tensor_size = new_response.tensor_sizes().empty()
+                                      ? 0
+                                      : new_response.tensor_sizes()[0];
         if (response.response_type() == new_response.response_type() &&
             response.devices() == new_response.devices() &&
-            dtype == new_entry.tensor->dtype() &&
+            response.tensor_type() == new_response.tensor_type() &&
             tensor_size + new_tensor_size <= TensorFusionThresholdBytes()) {
           // These tensors will fuse together well.
           tensor_size += new_tensor_size;
           response.add_tensor_name(new_response.tensor_names()[0]);
+          LOG(DEBUG) << "Fused tensor " << new_response.tensor_names()[0]
```

Comment:
I think we should either remove this or add it for the Allgather case as well.  I would lean towards removing, personally.

Reply:
Removed.